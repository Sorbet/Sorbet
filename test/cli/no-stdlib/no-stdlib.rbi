# typed: core

class Object; end

class String < Object
  sig do
    params(
        arg0: String,
    )
    .returns(String)
  end
  def *(arg0); end
end
