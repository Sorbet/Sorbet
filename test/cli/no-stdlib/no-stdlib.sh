#!/bin/bash
main/sorbet --silence-dev-message -e '"Foo" * "Foo"' 2>&1
main/sorbet --silence-dev-message --no-stdlib -e '"Foo" * "Foo"' test/cli/no-stdlib/no-stdlib.rbi 2>&1
