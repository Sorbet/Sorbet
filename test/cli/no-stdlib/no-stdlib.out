-e:1: `String("Foo")` doesn't match `Integer` for argument `arg0` https://sorbet.org/docs/error-reference#7002
     1 |"Foo" * "Foo"
        ^^^^^^^^^^^^^
    https://github.com/stripe/sorbet/tree/master/rbi/core/string.rbi#L15: Method `String#*` has specified `arg0` as `Integer`
    15 |        arg0: Integer,
                ^^^^
  Got String("Foo") originating from:
    -e:1:
     1 |"Foo" * "Foo"
                ^^^^^
Errors: 1
No errors! Great job.
